rust/src/r_api.rs

```diff
@@ -1,405 +1,529 @@
-use crate::ffi::{
```

Comment:
Can you explain what is the architectural decision behind the `ffi.rs` file and the `r_api.rs` file? I sort of see them as both providing similar things and not sure what the distinction is between the two.

Reply:
r_api.rs is the api that is exposed in Flutter. It uses ffi.rs (bdk-ffi). It does some structural wrapping to accommodate calls and code patterns that may not directly work from flutter if used with ffi.rs directly.
Also, adding some code to r_api(and root.dart) and removing from ffi.rs helps reduce the binary size. For example for txBuilder.